conan/tools/meson/toolchain.py

```diff
@@ -309,7 +357,7 @@ def _resolve_apple_flags_and_variables(self, build_env, compilers_by_conf):
             return
         # Calculating the main Apple flags
         min_flag, arch_flag, isysroot_flag = (
-            resolve_apple_flags(self._conanfile, is_cross_building=bool(self.cross_build)))
+            resolve_apple_flags(self._conanfile, is_cross_building=self._is_cross_building))
```

Comment:
I am concerned that the logic has changed. 
What if users are defining ``toolchain.cross_build = ...`` as it is public, and relying on the fact that ``MesonToolchain`` was considering it as cross-building?

Reply:
You're completely right. Let's rely on that attribute instead.

Reply:
I just verified that it's not breaking at all. All the checks using `self.cross_build` is at initialization time, so if the user changes the `self.cross_build` attr, it does not change anything.

Reply:
Keeping that one just in case 😉